src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchElementViewModel.cs

```diff
@@ -21,24 +21,25 @@ public class PackageManagerSearchElementViewModel : BrowserItemViewModel, IEquat
         public ICommand VisitRepositoryCommand { get; set; }
         public ICommand DownloadLatestToCustomPathCommand { get; set; }
 
-        /// <summary>
-        /// A function used to determine whether the user has installed this package already.
-        /// Matching occurs via package name.
-        /// </summary>
-        private Func<string,bool> isPackageInstalled;
-
         public new PackageManagerSearchElement Model { get; internal set; }
 
-
-        public PackageManagerSearchElementViewModel(PackageManagerSearchElement element, bool canLogin) : base(element)
+        /// <summary>
+        /// Alternative constructor which takes a Function object in order to
+        /// assist communication between the PackageManagerSearchViewModel
+        /// and the PackageManagerSearchElementViewModel.
+        /// </summary>
+        /// <param name="element">A PackageManagerSearchElement</param>
+        /// <param name="canLogin">A Boolean used for access control to certain internal packages.</param>
+        /// <param name="canInstall">Whether a package can be installed.</param>
+        internal PackageManagerSearchElementViewModel(PackageManagerSearchElement element, bool canLogin, bool canInstall) : base(element)
         {
             this.Model = element;
-
-            this.ToggleIsExpandedCommand = new DelegateCommand(() => this.Model.IsExpanded = !this.Model.IsExpanded );
+            
+            this.ToggleIsExpandedCommand = new DelegateCommand(() => this.Model.IsExpanded = !this.Model.IsExpanded);
 
             this.DownloadLatestCommand = new DelegateCommand(
                 () => OnRequestDownload(Model.Header.versions.Last(), false),
-                () => !Model.IsDeprecated && !IsInstalled);
+                () => !Model.IsDeprecated && CanInstall);
```

Comment:
Should this be `canInstall` or `CanInstall` after assigning it to `canInstall`?

Reply:
CanInstall is a wrapper over canInstall.
Not sure if there is a benefit to using the private property here ...Can you explain a little bit more what you mean ?

Reply:
You're using the old value of `CanInstall`, not the value of `canInstall` that is passed in, why?

Reply:
Also, there is the public property, `CanInstall`, the private backing field, `canInstall`, and a function parameter, also called `canInstall`. Perhaps you might want to change the parameter name to something else to avoid confusion.

Reply:
got it. Changed